### PR TITLE
style(Extension): fixing coding style in Extension step 1

### PR DESCRIPTION
**Pull request by Yassir-BenBOUBKER**
## Description

change the classComponent to FuncComponent.
Make the Extension Code Structure the same as in Resources.
change the Filter and search bar design 

Extension page Image
image

For update all and install all buttons
when there is an extension to install
image

when all the extension are installed
image

**Fixes** # (issue)
Bug: if the user start updating / installing all and then open deltails modal the other extension that aren't updated/ installed entered a loading loop
## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/front_src/src/Extensions/ExtensionDeleteDialog/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  onConfirm,
+  onCancel,
+}): JSX.Element => {
+  return (
+    <Dialog
+      open
+      labelConfirm="Delete"
+      onCancel={onCancel}
+      onClose={onCancel}
+      onConfirm={(): void => onConfirm(deletingEntity.id, deletingEntity.type)}
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDeleteDialog/index.tsx:22]: You can extract this in a specific function

**Comment by Yassir-BenBOUBKER** [comment]
This still some part that need to be added
like changing the button text from update all to update selection 
...

File: www/front_src/src/Extensions/ExtensionDeleteDialog/index.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    >
+      <Typography variant="h6">{deletingEntity.description}</Typography>
+
+      <Alert severity="warning">
+        Do you want to delete this extension? This action will remove all
+        associated data. 0015
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDeleteDialog/index.tsx:28]: Why 0015 ?

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
+/* eslint-disable react/jsx-filename-extension */
+/* eslint-disable react/prop-types */
+/* eslint-disable react/prefer-stateless-function */
+
+import React, { useState, useEffect } from 'react';
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:5]:
```suggestion
import * as React from 'react';
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 4]
```diff
@@ -0,0 +1,217 @@
+/* eslint-disable react/jsx-filename-extension */
+/* eslint-disable react/prop-types */
+/* eslint-disable react/prefer-stateless-function */
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:3]:
```suggestion
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+}: Props): JSX.Element | null => {
+  const [extensionDetails, setExtensionDetails] = useState<Entity | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const { sendRequest: sendExtensionDetailsValueRequests } =
+    useRequest<ExtensionDetails>({
+      request: getData,
+    });
+
+  useEffect(() => {
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:59]:
```suggestion
  React.useEffect(() => {
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+const ExtensionDetailPopup = ({
+  id,
+  type,
+  onCloseClicked,
+  onDeleteClicked,
+  onUpdateClicked,
+  onInstallClicked,
+}: Props): JSX.Element | null => {
+  const [extensionDetails, setExtensionDetails] = useState<Entity | null>(null);
+  const [loading, setLoading] = useState<boolean>(true);
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:52]:
```suggestion
  const [extensionDetails, setExtensionDetails] = React.useState<Entity | null>(null);
  const [loading, setLoading] = React.useState<boolean>(true);
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+  ContentSkeleton,
+  ReleaseNoteSkeleton,
+} from './LoadingSkeleton';
+
+interface Props {
+  id: string;
+  onCloseClicked: () => void;
+  onDeleteClicked: (id: string, type: string, description: string) => void;
+  onInstallClicked: (id: string, type: string) => void;
+  onUpdateClicked: (id: string, type: string) => void;
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:34]:
```suggestion
  onClose: () => void;
  onDelete: (id: string, type: string, description: string) => void;
  onInstall: (id: string, type: string) => void;
  onUpdate: (id: string, type: string) => void;
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+              ) : (
+                <Carousel
+                  fullHeightHover
+                  NextIcon={<ChevronRightIcon />}
+                  PrevIcon={<ChevronLeftIcon />}
+                  animation="slide"
+                  autoPlay={false}
+                >
+                  {extensionDetails.images
+                    ? extensionDetails.images.map((image) => (
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:104]:
```suggestion
                  {extensionDetails.images?.map((image) => (
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+                        <img
+                          alt={image}
+                          key={image}
+                          src={image}
+                          width={width}
+                        />
+                      ))
+                    : null}
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:112]:
```suggestion
}
```

File: www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+              size="small"
+              startIcon={<DeleteIcon />}
+              variant="contained"
+              onClick={(): void => {
+                onDeleteClicked(
+                  extensionDetails.id,
+                  extensionDetails.type,
+                  extensionDetails.title,
+                );
+              }}
```
Comment (Thebarda) [inline www/front_src/src/Extensions/ExtensionDetailsDialog/index.tsx:143]: Extract this into a specific function

File: www/front_src/src/Extensions/ExtensionDeleteDialog/index.tsx (hunk 2 above)
Comment (Yassir-BenBOUBKER) [reply www/front_src/src/Extensions/ExtensionDeleteDialog/index.tsx:28]: désolé, j'ai essayé de tester si mes modifications sont exécutées.

File: www/front_src/src/Extensions/models.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+export interface Extensions {
+  module: {
+    entities: Array<Entity>;
+  };
+  widget: {
+    entities: Array<Entity>;
+  };
+}
+
+export interface sendExtensionsRequestsType {
```
Comment (bdauria) [inline www/front_src/src/Extensions/models.ts:41]:
```suggestion
export interface ExtensionResult {
```

File: www/front_src/src/Extensions/models.ts [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  result: Extensions | string;
+  status: boolean;
+}
+
+export interface sendDeleteExtensionRequestsType {
```
Comment (bdauria) [inline www/front_src/src/Extensions/models.ts:46]:
```suggestion
export interface DeleteExtensionResult {
```

File: www/front_src/src/Extensions/models.ts [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+  result: string | null;
+  status: boolean;
+}
+
+
+export interface sendUpdateOrInstallExtensionRequestsTypes {
```
Comment (bdauria) [inline www/front_src/src/Extensions/models.ts:52]:
```suggestion
export interface InstallOrUpdateExtensionResult {
```

File: www/front_src/src/Extensions/Listing/index.tsx [hunk 14]
```diff
@@ -127,12 +130,46 @@ const ExtensionsManager = ({ reloadNavigation }): JSX.Element => {
...
+
+    const notUpdatableExtensionWidgetExiste = !isEmpty(
+      filter(pathEq(['version', 'outdated'], true), extensions.widget.entities),
+    );
+
+    setUpdatable(
+      notUpdatableExtensionModuleExiste || notUpdatableExtensionWidgetExiste,
+    );
+  }, [extensions]);
+
```
Comment (bdauria) [inline www/front_src/src/Extensions/Listing/index.tsx:172]: I would'nt use states here but simply variable to optimize the render cycles

Comment (Yassir-BenBOUBKER) [reply www/front_src/src/Extensions/Listing/index.tsx:172]: can i leave useEffect ???

**Pull request closed without merging**